### DISCUSSION TITLE
IMU to Camera constraint?

### DISCUSSION THREAD

**Discussion by lowellm91** [borglab/gtsam#1739] [Q&A] [answered]
Hi, I have an IMU and a Stereo Camera and some calibration information, I'm a little bit confused on how you create constraints between these two sensors. It could be modeled as a pose constraint with a really low variance but I would rather enforce an equality constraint between them(ex IMU_pose_(t_1) = T*(cam_pose_(t_1))

**Comment by ProfFan** [comment]
I think you can use the body_T_sensor of the IMUFactor

**Reply by lowellm91** [reply]
Hi Thanks, I was looking at the documentation and that doesn't appear to exist.

https://gtsam.org/doxygen/4.0.0/a03467.html

I must be missing something.

**Reply by scott-robotics** [reply, marked as answer]
The local transform / extrinsic between the sensors is handled in the Preintegration routine (via `PreintegratedRotationParams`).

```c++
gtsam::imuBias::ConstantBias prior_imu_bias;  // assume zero initial bias

auto parameters = gtsam::PreintegratedCombinedMeasurements::Params::MakeSharedD(0.0);
parameters->setBodyPSensor(local_transform); // <- Put your calibration info here
auto preintegration = std::make_shared<gtsam::PreintegratedImuMeasurements>(parameters, prior_imu_bias);
```

Here's an example using IMU preintegration: https://github.com/borglab/gtsam/blob/develop/examples/ImuFactorsExample.cpp
